swap_meet/vendor.py

```diff
@@ -1,2 +1,73 @@
 class Vendor:
-    pass
\ No newline at end of file
+
+    def __init__(self, inventory=[]):
+        self.inventory = inventory
+    
+    def add(self, item):
+        if not self.inventory:
+            self.inventory = []
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if not item in self.inventory:
+            return False
+        self.inventory.remove(item)
+        return item
+
+    def get_by_category(self, category):
+        category_list = []
+        for item in self.inventory:
+            if item.category == category:
+                category_list.append(item)
+        return category_list
+
+    def swap_items(self, other_vendor, my_item, other_vendor_item):
+        if not my_item in self.inventory or\
+            not other_vendor_item in other_vendor.inventory:
+            return False
+        my_item_to_swap = self.remove(my_item)
+        other_vendor.add(my_item_to_swap)
+        their_item_to_swap = other_vendor.remove(other_vendor_item)
+        self.add(their_item_to_swap)
+        return True
+
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        my_first_item = self.inventory[0]
+        other_vendor_first_item = other_vendor.inventory[0]
+        return self.swap_items(other_vendor, my_first_item,\
```

Comment:
A quick note about time complexity - It is great to reuse methods we've already written, however swap_items has linear time complexity because it invokes the remove() method.

Can you think of how you could write a custom swapping implementation to bring the time complexity to O(1) ? Recall that Python has syntax for reassigning values like `a, b = b, a`